### ISSUE TITLE
jax.scipy.linalg.toeplitz incorrectly propagates NaN to the entire output matrix

### ISSUE THREAD

**Issue by ALinrunrun** [jax-ml/jax#38636]
### Description

`jax.scipy.linalg.toeplitz` appears to incorrectly propagate `NaN` values when constructing a Toeplitz matrix from inputs containing nonfinite values.

A Toeplitz matrix should be constructed by selecting entries from the first column and first row. In this example, `row[0]` is `NaN`, but it should only affect positions that actually correspond to the overlapping diagonal value. It should not contaminate the whole output matrix.

However, JAX returns an all-`NaN` matrix, while the expected output contains finite values, `inf`, and `-0.0` in positions selected from `column` and `row`.

This looks like the implementation may be using arithmetic or masking in a way that allows `NaN` to propagate, instead of performing pure indexing/selection.

### Minimal reproducible example

```
#!/usr/bin/env python3
import os
import sys
import warnings

warnings.filterwarnings("ignore")
os.environ.setdefault("JAX_PLATFORMS", "cpu")

try:
    import numpy as np
    import jax.numpy as jnp
    import jax.scipy.linalg as jlinalg
except ImportError as e:
    print(f"missing dep: {e}")
    sys.exit(2)

column = np.array([np.nan, np.inf, -0.0], dtype=np.float32)
row = np.array([np.nan, 2.0, 3.0], dtype=np.float32)

expected = np.array(
    [
        [np.nan, 2.0, 3.0],
        [np.inf, np.nan, 2.0],
        [-0.0, np.inf, np.nan],
    ],
    dtype=np.float32,
)

out = np.asarray(jlinalg.toeplitz(jnp.asarray(column), jnp.asarray(row)))

print("jax:", out.tolist())
print("expected:", expected.tolist())

if not np.array_equal(out, expected, equal_nan=True):
    sys.exit(0)
sys.exit(1)
```

### Actual result

```
jax: [[nan, nan, nan], [nan, nan, nan], [nan, nan, nan]]
expected: [[nan, 2.0, 3.0], [inf, nan, 2.0], [-0.0, inf, nan]]
```

### Expected result

```
[[nan, 2.0, 3.0],
 [inf, nan, 2.0],
 [-0.0, inf, nan]]
```

For a Toeplitz matrix, the output should be formed by selecting values from `column` and `row`. `NaN` values should only appear in positions that correspond to an actual selected `NaN` input element. They should not propagate through unrelated finite, infinite, or signed-zero entries.

### System info (python version, jaxlib version, accelerator, etc.)

jax: 0.10.1
jaxlib: 0.10.1
numpy: 2.4.6
python: 3.13.13
platform: Ubuntu 24.04.3 LTS, Linux x86_64
accelerator: CPU backend

Note: GPU is present, but CUDA-enabled jaxlib is not installed, so JAX falls back to CPU.

**Comment by jakevdp** [comment]
This falls under #38651; please note that it may take a while for this to receive attention from the team.
